payout-gateway.php

```diff
@@ -90,256 +87,239 @@ function wc_payout_gateway_plugin_links( $links ) {
-
-			if ($debug == "yes") {
+			if ( $debug == 'yes' ) {
 				$logger = wc_get_logger();
-				$logger->log( 'payout_log', 'NOTIFICATION: ' . json_encode($notification)  );
+				$logger->log( 'payout_log', 'NOTIFICATION: ' . json_encode( $notification ) );
 			}
 
+			$config = array(
+				'client_id'     => $this->get_option( 'client_id' ),
+				'client_secret' => $this->get_option( 'client_secret' ),
+				'sandbox'       => $this->get_option( 'sandbox' ),
+			);
 
-		
+			// Initialize Payout
+			$payout = new Client( $config );
 
-		    $config = array(
-		        'client_id' => $this->get_option( 'client_id' ),
-		        'client_secret' => $this->get_option( 'client_secret' ),
-		        'sandbox' => $this->get_option( 'sandbox' )
-		    );
-
-		    // Initialize Payout
-		    $payout = new Client($config);
-
-
-
-		    $external_id = $notification->external_id;
-		    $store_payout_order_status = $notification->data->status;
-		    $payout_checkout_id = $notification->data->id;
-
-
-
-			if (isset($external_id) && isset($store_payout_order_status)) {
-		   		if (!$payout->verifySignature(array($external_id, $notification->type, $notification->nonce), $notification->signature)) {
-
- 					$result = array('error' => "Bad signature");
-					header_remove();	    
-					header('Content-Type: application/json');  
-					http_response_code(401);
-					echo json_encode($result, true);    
-					exit();
-
-
-		    } 
-
-
-		   	$order = wc_get_order( $external_id);
-			
-		  	update_post_meta($external_id, 'payout_order_status', $store_payout_order_status);
-		  	update_post_meta($external_id, 'payout_checkout_id', $payout_checkout_id);
-
-		  	$current_order_status = $order->get_status();
-
-
-		  	if ($debug == "yes") {
-				$logger->log( 'payout_log', 'STATUS: ' . json_encode($store_payout_order_status)  );
+			// check notification parameter presence
+			if ( ! isset( $notification->external_id, $notification->type, $notification->nonce, $notification->signature ) ) {
+				$this->notification_bail( 'Missing parameter' );
+				exit();
 			}
 
-		  	
-  
-			if ($store_payout_order_status == "succeeded") { 
-
-		
-	
-			    $order->payment_complete();
-
+			// check notification signature
+			$signature_data = array( $notification->external_id, $notification->type, $notification->nonce );
+			if ( ! $payout->verifySignature( $signature_data, $notification->signature ) ) {
+				$this->notification_bail( 'Bad signature' );
+				exit();
+			}
 
-			} else if ($store_payout_order_status == "expired" && $current_order_status != "completed") {
+			// process payment token notification
+			if ( 'payu_token.created' === $notification->type && isset( $notification->data->token_value ) ) {
+				$subscription_ids = array_keys( wcs_get_subscriptions_for_order( $notification->external_id ) );
+				foreach ( $subscription_ids as $subscription_id ) {
+					update_post_meta( $subscription_id, '_payout_payment_token', $notification->data->token_value );
```

Comment:
Jednou z veci, ktore som spominal na calle bolo to, ze token je velmi citliva vec a nebol by som rad keby sa ukladal len tak do DB. Je mozne ho ukladat bezpecnejsie?

Reply:
Je mozne token sifrovat a ukladat do databazy sifrovany.